### PR TITLE
Add prometheus operator ServiceMonitor samples

### PR DESCRIPTION
**Pull request by howardjohn**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: samples/addons/extras/prometheus-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+  selector:
+    matchExpressions:
+    - {key: istio, operator: In, values: [pilot]}
+  namespaceSelector:
+    any: true
+  endpoints:
+  - port: http-monitoring
+    interval: 15s
+---
+apiVersion: monitoring.coreos.com/v1
```
Comment (douglas-reid) [inline samples/addons/extras/prometheus-operator.yaml:51]: do we want to provide the k8s stuff?  or point people to the place to get them?

**Review by douglas-reid — COMMENTED** [review]
Just one question about scope...

File: samples/addons/extras/prometheus-operator.yaml (hunk 1 above)
Comment (howardjohn) [reply samples/addons/extras/prometheus-operator.yaml:51]: Forgot I was going to ask you about this. We probably should ignore it. Is there a good source to link to so the dashboards work? I'll take a look. This is slightly more advanced use case anyways so it doesn't need to be as streamlined. I heard some users conplain about double scraping kube so seems reasonable to remove

Comment (richardwxn) [reply samples/addons/extras/prometheus-operator.yaml]: I think it would be helpful to add links/pointers to k8s monitors configs.

Comment (howardjohn) [reply samples/addons/extras/prometheus-operator.yaml]: I added it to the README

**Review by richardwxn — APPROVED** [review]

**Pull request merged**